File: ecom-ai-studio/src/api/team.ts
/**
 * 团队管理 API
 * 封装团队创建、加入、查询、邀请码管理、成员管理接口调用
 */
import { api } from './client'
import type { ApiResponse, Team, TeamInvitation, InvitationVerification, TeamMember, TransferLog, PaginatedResponse } from '@/types'

// ==================== 团队管理 ====================

export async function createTeam(data: {
  name: string
  category: string
}): Promise<ApiResponse<{ team: Team }>> {
  return api.post<ApiResponse<{ team: Team }>>('/teams', data)
}

export async function joinTeam(inviteCode: string): Promise<ApiResponse<{ team: Team }>> {
  return api.post<ApiResponse<{ team: Team }>>('/teams/join', { inviteCode })
}

export async function getMyTeams(): Promise<ApiResponse<{ teams: Team[] }>> {
  return api.get<ApiResponse<{ teams: Team[] }>>('/teams/mine')
}

// ==================== 邀请码管理 ====================

export async function getTeamInvitation(
  teamId: string,
): Promise<ApiResponse<{ invitation: TeamInvitation | null }>> {
  return api.get<ApiResponse<{ invitation: TeamInvitation | null }>>(
    `/teams/${teamId}/invitation`,
  )
}

export async function refreshInvitation(
  teamId: string,
): Promise<ApiResponse<{ invitation: TeamInvitation }>> {
  return api.post<ApiResponse<{ invitation: TeamInvitation }>>(
    `/teams/${teamId}/invitation/refresh`,
  )
}

export async function verifyInvitation(
  code: string,
): Promise<ApiResponse<{ verification: InvitationVerification }>> {
  return api.post<ApiResponse<{ verification: InvitationVerification }>>(
    '/teams/invitation/verify',
    { code },
  )
}

// ==================== 团队成员管理 ====================

export async function getTeamMembers(
  teamId: string,
): Promise<ApiResponse<{ members: TeamMember[] }>> {
  return api.get<ApiResponse<{ members: TeamMember[] }>>(`/teams/${teamId}/members`)
}

// ==================== 解散团队 ====================

export async function dissolveTeam(teamId: string): Promise<ApiResponse<{
  transferredAmount: number
  message: string
}>> {
  return api.delete<ApiResponse<{ transferredAmount: number; message: string }>>(`/teams/${teamId}`)
}

// ==================== 转账功能 ====================

export async function transferToTeam(teamId: string, amount: number): Promise<ApiResponse<{
  amount: number
  newTeamBalance: number
  newPersonalBalance: number
}>> {
  return api.post<ApiResponse<{
    amount: number
    newTeamBalance: number
    newPersonalBalance: number
  }>>(`/teams/${teamId}/transfer`, { amount })
}

export async function getTeamTransferLogs(
  teamId: string,
  page?: number,
): Promise<PaginatedResponse<TransferLog[]> & ApiResponse<{ logs: TransferLog[] }>> {
  return api.get<PaginatedResponse<TransferLog[]> & ApiResponse<{ logs: TransferLog[] }>>(
    `/teams/${teamId}/transfer-logs`,
    { page, pageSize: 20 },
  )
}

File: auth-module-backend/routes/team.py
"""
团队路由模块
注册团队创建、加入、查询、邀请码管理、成员管理路由
"""
from flask import Blueprint
from controllers.team_controller import (
    create_team,
    join_team,
    get_my_teams,
    get_invitation,
    refresh_invitation,
    verify_invitation,
    get_team_members,
    dissolve_team,
    transfer_to_team,
    get_team_transfer_logs,
)
from middleware.auth_middleware import token_required

team_bp = Blueprint('team', __name__, url_prefix='/api/v1')

# 团队创建
team_bp.route('/teams', methods=['POST'])(token_required(create_team))

# 通过邀请码加入团队
team_bp.route('/teams/join', methods=['POST'])(token_required(join_team))

# 获取当前用户的团队列表
team_bp.route('/teams/mine', methods=['GET'])(token_required(get_my_teams))

# ---- 邀请码管理（/teams/invitation/verify 必须在 /teams/<int:team_id> 之前注册） ----

# 验证邀请码
team_bp.route('/teams/invitation/verify', methods=['POST'])(token_required(verify_invitation))

# 查看团队邀请码（owner/admin）
team_bp.route('/teams/<int:team_id>/invitation', methods=['GET'])(token_required(get_invitation))

# 刷新邀请码（owner/admin）
team_bp.route('/teams/<int:team_id>/invitation/refresh', methods=['POST'])(token_required(refresh_invitation))

# ---- 团队成员管理 ----

# 获取团队成员列表
team_bp.route('/teams/<int:team_id>/members', methods=['GET'])(token_required(get_team_members))

# ---- 解散团队（owner only） ----

team_bp.route('/teams/<int:team_id>', methods=['DELETE'])(token_required(dissolve_team))

# ---- 转账功能 ----

# 个人向团队转账
team_bp.route('/teams/<int:team_id>/transfer', methods=['POST'])(token_required(transfer_to_team))

# 获取团队转账日志
team_bp.route('/teams/<int:team_id>/transfer-logs', methods=['GET'])(token_required(get_team_transfer_logs))